cernopendata/templates/cernopendata_records_ui/records/detail.html

```diff
@@ -252,6 +294,65 @@ <h4> Export </h4>
     }
   );
 
+  recordApp.controller(
+    'indexFileListModalCtrl',
+    [
+      '$scope',
+      '$interpolate',
+      '$http',
+      '$uibModal',
+      function($scope,  $interpolate , $http, $uibModal) {
+        var $ctrl = this;
+
+        $ctrl.openModal = function (pid, file) {
+          var modalInstance = $uibModal.open({
+            animation: $ctrl.animationsEnabled,
+            ariaLabelledBy: 'modal-title',
+            ariaDescribedBy: 'modal-body',
+            templateUrl: "/static/templates/indexFileListModal.html",
+            controller: 'indexFileListModalInstanceCtrl',
+            controllerAs: '$ctrl',
+            windowClass: 'index-file-list-modal',
+            resolve: {
+              $http: function() {
+                return $http;
+              },
+              pid: function() {
+                return pid;
+              },
+              uri: function() {
+                if (file.endsWith('index.txt')) {
```

Comment:
You can update this now as per new `type` values of "index.txt" and "index.json"